src/libs/config/yaml.cpp

```diff
@@ -583,6 +584,26 @@ abs_cfg_path(const std::string &path)
 	}
 }
 
+/** Replace $host in string with hostname
+ * @param prelim preliminary filename (potentially with $host)
+ * @return filename with $host replaced with hostname
+ */
+static std::string
+insert_hostname(std::string prelim)
+{
+	static char *hostname = NULL;
+	if (hostname == NULL) {
+		hostname = new char[256];
+		gethostname(hostname, 256);
+	}
+	size_t repl_position = prelim.find("$host");
+	if (repl_position == std::string::npos) {
+		return prelim;
+	} else {
+		return prelim.replace(repl_position, 2, std::string(hostname));
```

Comment:
The `2` is wrong, that's the length of `<>`.

It would be best to define a `const std::string to_replace = "$host"` and then use that everywhere to avoid such bugs.

Reply:
Ah, sorry for the sloppy correction.